module_checksum_upgrade/tests/test_module.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# Copyright 2018 ACSONE SA/NV.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import logging
+import os
+import tempfile
+
+from openerp.modules import get_module_path
+from openerp.tests.common import TransactionCase
+
+_logger = logging.getLogger(__name__)
+try:
+    from checksumdir import dirhash
+except ImportError:
+    _logger.debug('Cannot `import checksumdir`.')
+
+
+class TestModule(TransactionCase):
+
+    def setUp(self):
+        super(TestModule, self).setUp()
+        module_name = 'module_checksum_upgrade'
+        self.own_module = self.env['ir.module.module'].search([
+            ('name', '=', module_name),
+        ])
+        self.own_dir_path = get_module_path(module_name)
+        self.own_checksum = dirhash(
+            self.own_dir_path,
+            'sha1',
+            excluded_extensions=['pyc', 'pyo'],
+        )
+        self.own_writeable = os.access(self.own_dir_path, os.W_OK)
+
+    def test_compute_checksum_dir(self):
+        """It should compute the directory's SHA-1 hash"""
+        self.assertEqual(
+            self.own_module._get_checksum_dir(), self.own_checksum,
+            'Module directory checksum not computed properly',
+        )
+
+    def test_compute_checksum_dir_ignore_excluded(self):
+        """It should exclude .pyc/.pyo extensions from checksum
+        calculations"""
+        if not self.own_writeable:
+            self.skipTest("Own directory not writeable")
```

Comment:
The original addon test does it differently.

Instead, it writes a false checksum before checking the upgrade behavior: https://github.com/OCA/server-tools/blob/83bce19b454041344ed79de7b7374a9f01f8e98f/module_auto_update/tests/test_module_upgrade.py#L24

It makes addon testable everywhere; please do that instead. :blush:

Reply:
This particular test comes unchanged from the original module:

https://github.com/OCA/server-tools/blob/3ed2a554b0465614ed75be4d4991333e7e2f98a5/module_auto_update/tests/test_module.py#L58-L59